cloudinit/sources/DataSourceWSL.py

```diff
@@ -143,22 +143,35 @@ def candidate_user_data_file_names(instance_name) -> List[str]:
     ]
 
 
-def load_yaml_or_bin(data_path: str) -> Optional[Union[dict, bytes]]:
-    """
-    Tries to load a YAML file as a dict, otherwise returns the file's raw
-    binary contents as `bytes`. Returns `None` if no file is found.
-    """
-    try:
-        bin_data = util.load_binary_file(data_path)
-        dict_data = util.load_yaml(bin_data)
-        if dict_data is None:
-            return bin_data
+class ConfigData:
+    """Models a piece of configuration data as a dict if possible, while
+    retaining its raw representation alongside its file path"""
 
-        return dict_data
-    except FileNotFoundError:
-        LOG.debug("No data found at %s, ignoring.", data_path)
+    def __init__(self, path: PurePath):
+        if not os.path.isfile(path.as_posix()):
+            raise FileNotFoundError(f"Config file {path} not found")
+
+        self._raw_data: str = util.load_binary_file(path).decode("utf-8")
```

Comment:
Indeed decoding is not necessary. I was puzzled with that because `self.userdata_raw` is declared in the superclass as `str`, but yeah, we can remove the decoding without any issues for the linters and tests in use as well as behavior.